src/main/java/uk/ac/cam/cl/dtg/segue/api/SessionValidator.java

```diff
@@ -0,0 +1,114 @@
+package uk.ac.cam.cl.dtg.segue.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.inject.Inject;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.container.ContainerResponseContext;
+import jakarta.ws.rs.container.ContainerResponseFilter;
+import jakarta.ws.rs.container.PreMatching;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.Cookie;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Provider;
+import org.jboss.resteasy.core.interception.jaxrs.ContainerResponseContextImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.segue.api.managers.UserAuthenticationManager;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Map;
+
+import static uk.ac.cam.cl.dtg.segue.api.Constants.*;
+
+@Provider
+@PreMatching
+public class SessionValidator implements ContainerRequestFilter, ContainerResponseFilter {
+
+    private static final Logger log = LoggerFactory.getLogger(SessionValidator.class);
+
+    private final UserAuthenticationManager userAuthenticationManager;
+    private final PropertiesLoader properties;
+    private final Integer sessionExpirySeconds;
+    @Context
+    private HttpServletRequest httpServletRequest;
+    @Context
+    private HttpServletResponse httpServletResponse;
+
+    @Inject
+    public SessionValidator(final UserAuthenticationManager userAuthenticationManager, PropertiesLoader properties) {
+        this.userAuthenticationManager = userAuthenticationManager;
+        this.properties = properties;
+        this.sessionExpirySeconds = getSessionExpirySeconds();
+    }
+
+    private Integer getSessionExpirySeconds() {
```

Comment:
This was the accessor method I meant in the comment on UserAuthenticationManager. If this method needs to be used by both classes perhaps extract it into a utility class.